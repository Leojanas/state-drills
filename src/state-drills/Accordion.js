import { render } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';

class Accordion extends React.Component {
    state = {
        currentTab: null
    }
    handleButtonClick(index) {
        if(this.state.currentTab === index){
            this.setState({
                currentTab: null
            })
        }else{
            this.setState({
                currentTab: index
            })
        }
    }
    renderContent(index){
        if(this.state.currentTab === index){
            return (
            <p>{this.props.sections[index].content}</p>
            )
        }else{
            return null
        }
    }
    render(){
        const items = this.props.sections.map((section, index) => {
        return(
        <li key={index}>
            <button key={index} onClick={() => {this.handleButtonClick(index)}}>
                {section.title}
            </button>
            {this.renderContent(index)}
        </li>
        )
        })
        return (
            <div>
                <ul>
                    {items}
                </ul>
            </div>
        )
    }
}

export default Accordion