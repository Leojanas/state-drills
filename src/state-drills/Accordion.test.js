import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ];

it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion sections={sections}/>, div);
    ReactDOM.unmountComponentAtNode(div);
    })

it('renders first with all sections closed', () => {
        const wrapper = shallow(<Accordion sections={sections}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
        })

it('opens first section when clicked', () => {
            const wrapper = shallow(<Accordion sections={sections} />)
            wrapper.find('button').at(0).simulate('click')
            expect(toJson(wrapper)).toMatchSnapshot()
          })

it('opens second section when clicked', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

it('only opens last section when multiple have been clicked', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(2).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

it('closes a section when clicked again', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })