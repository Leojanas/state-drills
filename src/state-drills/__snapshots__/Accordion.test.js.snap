// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`closes a section when clicked again 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`only opens last section when multiple have been clicked 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`opens first section when clicked 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`opens second section when clicked 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`renders first with all sections closed 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
